forum/backends/mysql/models.py

```diff
@@ -0,0 +1,576 @@
+
+        return {
+            "_id": str(self.pk),
+            "votes": votes,
+            "visible": self.visible,
+            "abuse_flaggers": self.abuse_flaggers,
+            "historical_abuse_flaggers": self.historical_abuse_flaggers,
+            "thread_type": self.thread_type,
+            "context": self.context,
+            "comment_count": self.comment_count,
+            "at_position_list": [],
+            "title": self.title,
+            "body": self.body,
+            "course_id": self.course_id,
+            "anonymous": self.anonymous,
+            "anonymous_to_peers": self.anonymous_to_peers,
+            "closed": self.closed,
+            "author_id": self.author.pk,
+            "author_username": self.author.username,
+            "updated_at": self.updated_at.isoformat() if self.updated_at else None,
+            "created_at": self.created_at.isoformat() if self.created_at else None,
+            "last_activity_at": (
+                self.last_activity_at.isoformat() if self.last_activity_at else None
+            ),
+            "edit_history": edit_history,
+        }
+
+    class Meta:
+        indexes = [
+            models.Index(fields=["context"]),
+            models.Index(fields=["author"]),
+            models.Index(fields=["author", "course_id"]),
+            models.Index(fields=["course_id", "anonymous", "anonymous_to_peers"]),
+            models.Index(
+                fields=["author", "course_id", "anonymous", "anonymous_to_peers"]
+            ),
+        ]
+
+
+class Comment(Content):
+    """Comment model class"""
+
+    endorsement: models.JSONField[dict[str, Any], dict[str, Any]] = models.JSONField(
+        default=dict
+    )
+    sort_key: models.CharField[Optional[str], str] = models.CharField(
+        max_length=255, null=True, blank=True
+    )
+    child_count: models.PositiveIntegerField[int, int] = models.PositiveIntegerField(
+        default=0
+    )
+    retired_username: models.CharField[Optional[str], str] = models.CharField(
+        max_length=255, null=True, blank=True
+    )
+    comment_thread: models.ForeignKey[CommentThread, CommentThread] = models.ForeignKey(
+        CommentThread, on_delete=models.CASCADE
+    )
+    parent: models.ForeignKey[Comment, Comment] = models.ForeignKey(
+        "self", on_delete=models.CASCADE, null=True, blank=True
+    )
+
+    def to_dict(self) -> dict[str, Any]:
+        """Return a dictionary representation of the model."""
+        votes = {
+            "up": [vote.user.pk for vote in self.votes.filter(vote=1)],
+            "down": [vote.user.pk for vote in self.votes.filter(vote=-1)],
+            "up_count": self.votes.filter(vote=1).count(),
+            "down_count": self.votes.filter(vote=-1).count(),
+            "count": self.votes.filter(vote=1).count()
+            - self.votes.filter(vote=-1).count(),
+            "point": self.votes.filter(vote=1).count()
+            - self.votes.filter(vote=-1).count(),
+        }
```

Comment:
Can we move this calculation to the content model? As we are using votes in both, threads and comments?

Reply:
Done